src/plugins/data/server/search/collectors/usage.ts

```diff
@@ -38,44 +40,53 @@ export interface SearchUsage {
   trackSuccess(duration: number): Promise<void>;
 }
 
-export function usageProvider(core: CoreSetup): SearchUsage {
+export function usageProvider(
+  core: CoreSetup,
+  initializerContext: PluginInitializerContext<ConfigSchema>
```

Comment:
is there potential to make this optional?

Reply:
I searched code base, there is only one reference that search service call usageProvider, it is better to keep the config pass from search service to usageprovider, otherwise, it will easy break some corner cases which related with the connection between config and service.

Reply:
discussed with Rocky offline, this is one time initial setup for search service. will keep current implement for now.